### PR TITLE
Update to 3.3

### PR DESCRIPTION
**Pull request by jschueller**
Closes https://github.com/conda-forge/opencv-feedstock/issues/71

### CODE REVIEW

**Comment by jakirkham** [comment]
I don't think Python 2.7 can be supported on Windows any more. Upstream has been pretty clear that have no interest in making things work with VS 2008 and it has been too much of a pain to fix. So we are going to have to drop Python 2.7 on Windows.

**Comment by jschueller** [comment]
@jakirkham do you have a link to support this to add in meta.yaml?

File: recipe/meta.yaml [hunk 1]
```diff
@@ -8,18 +8,12 @@ package:
...
-    - opencv_drawing_abs.patch              # [win and py<35]
-    - opencv_drawing_data_member.patch      # [win and py<35]
-    - opencv_core_int64_abs.patch           # [win and py<35]
+  sha256: 8bb312b9d9fd17336dc1f8b3ac82f021ca50e2034afc866098866176d985adc6
 
 build:
-  number: 204
+  number: 200
+  # Python2.7 support dropped: https://github.com/opencv/opencv/issues/8481
+  skip: true  # [win and py27]
```
Comment (jakirkham) [inline recipe/meta.yaml]: xref: https://github.com/opencv/opencv/issues/8481

Comment (jschueller) [reply recipe/meta.yaml]: thanks

**Comment by jakirkham** [comment]
We should switch to the [new way of handling NumPy]( https://conda-forge.org/docs/meta.html#building-against-numpy ) and re-render. This should cutdown on the CI resources required to build OpenCV.

**Comment by jschueller** [comment]
ok, I'm adding this

**Comment by jschueller** [comment]
these do not seem up to date

**Comment by jakirkham** [comment]
Basically should look like this.

```yaml
  build:
    - numpy 1.8.*  # [not (win and (py35 or py36))]
    - numpy 1.9.*  # [win and (py35 or py36)]
  run:
    - numpy >=1.8  # [not (win and (py35 or py36))]
    - numpy >=1.9  # [win and (py35 or py36)]
```

**Comment by jakirkham** [comment]
> these do not seem up to date

Right. So we are trying to build on an old NumPy to ensure that a wide range of NumPy versions are supported. The goal was to match wheels more closely, but we were unable to match their versions exactly and it came with a lot of overhead (different NumPy versions for different Python versions). This is close enough to what we want. Though we would really like it if Windows diverged less.

**Comment by jschueller** [comment]
I meant their were older than the minimum (1.11) of the previous build matrix so I wanted to be sure.
Is this documented somewhere ?

**Comment by jakirkham** [comment]
Yep, it's true they are older. Thanks for checking.

It is. The link I gave you is to our docs page on the website.

**Comment by jschueller** [comment]
oh, right

**Comment by jschueller** [comment]
@jakirkham conda cannot resolve numpy 1.9.*  and py36

File: recipe/meta.yaml [hunk 2]
```diff
@@ -35,16 +29,17 @@ requirements:
...
     - jasper                       # [unix]
     - zlib 1.2.*
     - jpeg 9*
     - libtiff 4.0.*
     - libwebp 0.5.*
-    - harfbuzz 1.3.*               # [unix]
-    - libpng >=1.6.23,<1.7
+    - harfbuzz 1.3.4               # [unix]
+    - libpng >=1.6.28,<1.7
     - msinttypes                   # [win and py<35]
```
Comment (jakirkham) [inline recipe/meta.yaml]: I guess we can do away with this.

**Comment by jakirkham** [comment]
> conda cannot resolve numpy 1.9.* and py36

Yeah I'm not seeing it in `defaults`. Can you please raise an issue on the org's webpage repo and ping core about this? Thanks.

**Comment by jschueller** [comment]
see https://github.com/ContinuumIO/anaconda-issues/issues/5720

File: recipe/meta.yaml (hunk 2 above)
Comment (jschueller) [reply recipe/meta.yaml]: But, that's the official pinning

Comment (jakirkham) [reply recipe/meta.yaml]: Was referring to `msinttypes`.

Comment (jschueller) [reply recipe/meta.yaml]: ok, I'll delete that.

**Comment by jakirkham** [comment]
With `pandas`, [this]( https://github.com/conda-forge/pandas-feedstock/blob/35fb07a65711f8eaf0432f2cbbebb5d2c216e039/recipe/meta.yaml#L22-L29 ) was done. Maybe we can do the same thing here?

```yaml
  build:
    - numpy 1.8.*   # [not (win and (py35 or py36))]
    - numpy 1.9.*   # [win and py35]
    - numpy 1.11.*  # [win and py36]

  run:
    - numpy >=1.8   # [not (win and (py35 or py36))]
    - numpy >=1.9   # [win and py35]
    - numpy >=1.11  # [win and py36]
```

**Comment by jschueller** [comment]
ok, updated

**Comment by hajapy** [comment]
Anyone know why there are not travis ci test builds for OS X?

**Comment by jschueller** [comment]
@jakirkham @hajapy ci has passed, this is ready to go from my point of view

**Comment by hajapy** [comment]
My bad, I see the travis ci tests now. I see no reason to hold this up, LGTM.

**Comment by jschueller** [comment]
cc @conda-forge/core

**Comment by jakirkham** [comment]
LGTM. @patricksnape?

**Comment by patricksnape** [comment]
Amazing work @jschueller

**Pull request merged**